### ISSUE TITLE
Unable to install django-oauth-toolkit==1.0.0 when using pip ... --no-binary :all: 

### ISSUE THREAD

**Issue by bastbnl** [evonove/django-oauth-toolkit#532]
```
root@bastb-vps:~# virtualenv test-dot
Running virtualenv with interpreter /usr/bin/python2
New python executable in /root/test-dot/bin/python2
Also creating executable in /root/test-dot/bin/python
Installing setuptools, pkg_resources, pip, wheel...done.
root@bastb-vps:~# source test-dot/bin/activate
(test-dot) root@bastb-vps:~# pip --version
pip 9.0.1 from /root/test-dot/local/lib/python2.7/site-packages (python 2.7)
(test-dot) root@bastb-vps:~# pip install django-oauth-toolkit --no-binary :all:
Collecting django-oauth-toolkit==1.0.0
  Could not find a version that satisfies the requirement django-oauth-toolkit==1.0.0 (from versions: 0.1.0, 0.2.0, 0.2.1, 0.3.0, 0.3.1, 0.3.2, 0.4.0, 0.4.1, 0.5.0, 0.6.0, 0.6.1, 0.7.0, 0.7.1, 0.7.2, 0.8.0, 0.8.1, 0.8.2, 0.9.0, 0.10.0, 0.11.0, 0.12.0)
No matching distribution found for django-oauth-toolkit==1.0.0
```
It is found, including dependencies, when I leave out `--no-binary :all:`:
```
(test-dot) root@bastb-vps:~# pip install django-oauth-toolkit==1.0.0
Collecting django-oauth-toolkit==1.0.0
  Using cached django_oauth_toolkit-1.0.0-py2.py3-none-any.whl
Collecting requests>=2.13.0 (from django-oauth-toolkit==1.0.0)
  Using cached requests-2.18.4-py2.py3-none-any.whl
Collecting oauthlib>=2.0.1 (from django-oauth-toolkit==1.0.0)
Collecting django>=1.10 (from django-oauth-toolkit==1.0.0)
  Using cached Django-2.0.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-9GXRVZ/django/setup.py", line 32, in <module>
        version = __import__('django').get_version()
      File "django/__init__.py", line 1, in <module>
        from django.utils.version import get_version
      File "django/utils/version.py", line 61, in <module>
        @functools.lru_cache()
    AttributeError: 'module' object has no attribute 'lru_cache'

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-9GXRVZ/django/
```
I'm using `--no-binary` to prevent platform mismatches with compiled code, as found in this ticket: 
https://github.com/paramiko/paramiko/issues/1023

I was reviewing the differences between 0.12.0 and 1.0.0 and noticed that there's a tarball named `django-oauth-toolkit-0.12.0.tar.gz`. I found only a wheel on pypi for django-oauth-toolkit==1.0.0. Would it be possible to add the tarball for 1.0.0 (and newer) in current and future releases?

**Comment by jleclanche** [comment]
You are depending on django 2.0 somewhere, which is Python 3 only, and you're using python 2...

Here is a clean way to install DOT:

1. Create a new virtualenv (python3 if possible)
2. Run `pip install --upgrade pip setuptools wheel` to make sure you have the latest version of all three.
3. Run `pip install django-oauth-toolkit`.

**Comment by bastbnl** [comment]
Oh, sorry, I see I didn't stress the issue enough. Please create a new virtualenv - using a python interpreter of your choice - and run this:
`pip install django-oauth-toolkit==1.0.0 --no-binary :all:` 

You will not be able to install it:
```
Collecting django-oauth-toolkit==1.0.0
  Could not find a version that satisfies the requirement django-oauth-toolkit==1.0.0 (from versions: 0.1.0, 0.2.0, 0.2.1, 0.3.0, 0.3.1, 0.3.2, 0.4.0, 0.4.1, 0.5.0, 0.6.0, 0.6.1, 0.7.0, 0.7.1, 0.7.2, 0.8.0, 0.8.1, 0.8.2, 0.9.0, 0.10.0, 0.11.0, 0.12.0)
No matching distribution found for django-oauth-toolkit==1.0.0
```

You will be able to install version 0.12.0 using `--no-binary :all:`, which I tried to show in the output, but that failed on the Django 2.0 version. I can see that's confusing.

**Comment by bastbnl** [comment]
@jleclanche I'm not sure if you got the previous update

**Comment by jleclanche** [comment]
Right, uhm, sorry, I had misread the issue.

It looks like we are not shipping source distributions anymore. But why is that an issue? The shipped wheel has no compiled code in it, it's just a zip.

**Comment by bastbnl** [comment]
No problem. I'm glad you found some time to review it once again.

Well, it's a problem for me because I populate the virtualenv using the same `pip` command line. I would run into the `manage.py shell` delays as found in issue https://github.com/paramiko/paramiko/issues/1023 since `pynacl` is included into the same virtualenv.

I guess `pip` will not select the wheel.

**Comment by jleclanche** [comment]
i would recommend installing paramiko separately using --no-binary, then install the rest of your dependencies.

**Comment by bastbnl** [comment]
That would certainly work around the issue when it comes to `pynacl` (and `paramiko`), but I would run into it again with another package. I'm old-skool and it feels better to build any binary code on the host that will run the software.

Is it hard to add the source distribution again, in a future release? It looks like a common practice.

**Comment by jleclanche** [comment]
The python community as a whole is moving more towards wheels than sdists (and I myself have stopped distributing anything else than wheels as well for pure python projects). I get you re. building the code on the host, but that only matters for cpython modules, so im still of the opinion of "install those first".

**Comment by bastbnl** [comment]
Thanks for clarifying again, @jleclanche.

Shipping both a wheel and a source distribution still appears to be the preferred way, as documented on https://packaging.python.org/tutorials/distributing-packages/#packaging-your-project: 
>  Minimally, you should create a Source Distribution

I did a quick scan on 10+ packages listed on https://pythonwheels.com/ and noticed that the packages I checked both have a wheel and a source package. The wheel distribution appears to be an addition to the source distribution.

So I guess I'm kindly asking to reconsider adding the source distribution. I totally understand you have a lot of work on your hands so I'll add a PR if that helps.
